### PR TITLE
Use updated symbolic_helper.check_training_mode

### PR DESCRIPTION
**Pull request by jingyanwangms**
**Description**: 
symbolic_helper._training_mode no longer exists, according to https://github.com/pytorch/pytorch/blob/master/torch/onnx/symbolic_helper.py#L62, the new api is symbolic_helper.check_training_mode
**Motivation and Context**
Gives module 'torch.onnx.symbolic_helper' has no attribute '_training_mode' right now

### CODE REVIEW

File: orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def _export_pt_1_10(g, n, *args, **kwargs):
                 "wrap exportable sub-nn.Module's as ORTModule."
             )
         inplace = kwargs["inplace"]
-        training_mode = symbolic_helper._training_mode
+        training_mode = symbolic_helper.check_training_mode
```
Comment (justinchuby) [inline orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: `_training_mode` is a private field not exposed to users. Do you need access to the current training mode during export?

Comment (ytaous) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: +1

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: Created https://github.com/pytorch/pytorch/issues/79950 cc @bowenbao

Comment (ytaous) [inline orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: >check_training_mode [](http://example.com/codeflow?start=40&length=19)

might be better to use this for now?
training_mode = symbolic_helper.GLOBALS.training_mode

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: For a temporary fix, id use torch.onnx._globals.GLOBAL.export_training` If there is anything to unblock.  https://github.com/pytorch/pytorch/blob/765b6a8fab02a55cf85058037f9ab89fb00a5b23/torch/onnx/_globals.py#L30 I do suggest leaving a todo and plan to update it once we expose the value properly. Do you have any suggestions on the api?

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: For the release version of PyTorch 1.12, you _may_ need  torch.onnx._globals.GLOBAL.training_mode

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: https://github.com/pytorch/pytorch/blob/734a97a7c828d6bfe82764c9cae399c349828091/torch/onnx/_globals.py#L28 From the 1.12 release branch: sounds right. Note that even though the field was marked as optional[…] it was really optional[bool]. The type was fixed after the 1.12 cut

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: Is there anyway to know the training mode and keep it when calling export()? This way we don’t need to rely on an internal state

Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: > Is there anyway to know the training mode and keep it when calling export()? This way we don’t need to rely on an internal state

Have you looked into https://github.com/pytorch/pytorch/blob/master/torch/onnx/utils.py#L77-L124?

Basically the exporter tries to respect whatever `model.training` has, which all `torch.nn.Module` provides. The `torch.onnx.export(..., mode: _C_onnx.TrainingMode, ...)` is used to determine what a default value is in some cases.

Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: > `_training_mode` is a private field not exposed to users. Do you need access to the current training mode during export?
> 
> (check_training_mode is a function which returns None. It doesn't seem fitting to be used here)

One alternative is simply convert it to public/property :)

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: I would like to be cautious about exposing internal states. Once we do that, changes will be very hard. I am hoping we remove more globals in the future.

Comment (jingyanwangms) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: @justinchuby @ytaous 
FYI it's `torch.onnx._globals.GLOBALS.training_mode` not `torch.onnx._globals.GLOBAL.training_mode` 
I'll update this PR

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:40]: good catch sorry about that

Comment (jingyanwangms) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py]: No worries. The PR is updated now

File: orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py [hunk 2]
```diff
@@ -37,7 +37,8 @@ def _export_pt_1_10(g, n, *args, **kwargs):
                 "wrap exportable sub-nn.Module's as ORTModule."
             )
         inplace = kwargs["inplace"]
-        training_mode = symbolic_helper._training_mode
+        # TODO move to public API once exporter team exposes that
+        training_mode = _globals.GLOBALS.training_mode
```
Comment (pengwa) [inline orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:41]: just confirm, is this compatible with older version PyTorch? If not, should we add torch version check here?

Comment (justinchuby) [reply orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py:41]: It is recently introduced. I would just extract it into a function (something like `get_exporter_training_mode`) and do a try catch there. There is a risk that we will need to handle arbitrarily many different cases to access this var because internal things can change many times in nightly builds.

**Review by ytaous — APPROVED** [review]

**Comment by AllenTiTaiWang** [comment]
Following up on this.
